test/list_test.rb

```diff
@@ -15,15 +15,12 @@
         list.length.must_equal 0
 
         list.add('foo')
-        list.length.must_equal 1
```

Comment:
Removing this check is probably a bit pedantic, but I like to test only the method under test and nothing more. Strictly speaking, you could have a correct `#add` method with an incorrect `#length` method. The two are actually orthogonal from the view of the person consuming the interface.

If the `#length` method is actually incorrect, I'd prefer the test for `#length` to fail instead of the test for `#add` failing as well.